Objects/longobject.c

```diff
@@ -5355,9 +5303,19 @@ long_rshift(PyObject *a, PyObject *b)
     if (_PyLong_IsZero((PyLongObject *)a)) {
         return PyLong_FromLong(0);
     }
-    if (divmod_shift(b, &wordshift, &remshift) < 0)
-        return NULL;
-    return long_rshift1((PyLongObject *)a, wordshift, remshift);
+    if (PyLong_AsUInt64(b, &shiftby) < 0) {
+        if (!PyErr_ExceptionMatches(PyExc_OverflowError)) {
+            return NULL;
+        }
+        PyErr_Clear();
+        if (_PyLong_IsNegative((PyLongObject *)a)) {
+            return PyLong_FromLong(-1);
+        }
+        else {
+            return PyLong_FromLong(0);
+        }
```

Comment:
`else` is superfluous.

```suggestion
        return PyLong_FromLong(0);
```

... or:

```suggestion
        return _PyLong_GetZero();
```

Reply:
But this will lose symmetry. Note that this code is a duplicate of https://github.com/python/cpython/pull/123498/files#diff-1a6e70e2beeecad88840c67284ac4d54a36998029244771fcc820e801390726aR5334-R5339